src/util/ConfigManager/ConfigManager.h

```diff
@@ -287,5 +353,78 @@ class ConfigManager {
   */
   auto configurationOptions();
   auto configurationOptions() const;
+
+  /*
+  @brief Call all the validators to check, if the current value is valid.
+  */
+  void verifyWithValidators() const;
+
+  /*
+  Checks, if the given configuration option is contained in the manager, or
+  any of it's sub manager.
+  Note: This checks for identity, not equality.
+  */
+  bool containsOption(const ConfigOption& opt) const;
+
+  /*
+  @brief The implementation of `addValidator` and `addOptionValidator`.
+
+  @param addValidatorFunctionName The name of the function, that is calling this
+  implementation. So that people can easier identify them.
+  @param translationFunction Transforms the given `ConfigOptionProxy` in the
+  value form for the validator function.
+  @param validatorFunction Should return true, if the transformed
+  `ConfigOptionProxy` represent valid values. For example: There configuration
+  options were all set at run time, or contain numbers bigger than 10.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param parameterProxy Proxies for the configuration options, who will be
+  passed to the validator function as function arguments, after being
+  transformed. Will keep the same order.
+  */
+  template <typename TranslationFunction, typename ValidatorFunction,
+            isInstantiation<ConstConfigOptionProxy>... ValidatorParameter>
+  requires(std::invocable<TranslationFunction, const ValidatorParameter> &&
+           ...) &&
+          Validator<ValidatorFunction,
+                    std::invoke_result_t<TranslationFunction,
+                                         ValidatorParameter>...> &&
+          (sizeof...(ValidatorParameter) > 0)
+  void addValidatorImpl(std::string_view addValidatorFunctionName,
+                        TranslationFunction translationFunction,
+                        ValidatorFunction validatorFunction,
+                        std::string_view errorMessage,
+                        const ValidatorParameter... parameterProxy) {
+    // Check, if we contain all the configuration options, that were given us.
+    auto checkIfContainOption = [this, &addValidatorFunctionName]<typename T>(
+                                    ConstConfigOptionProxy<T> opt) {
+      if (!containsOption(opt.getConfigOption())) {
+        throw std::runtime_error(absl::StrCat(
+            "Error while adding validator with ", addValidatorFunctionName,
+            ": The given configuration "
+            "option '",
+            opt.getConfigOption().getIdentifier(),
+            "' is not contained in the configuration manager, or its sub "
+            "managers."));
```

Comment:
```suggestion
            "' is not contained in the configuration manager or any of its sub "
            "managers."));
```

Reply:
Thank you, I completely overlooked that mention of sub mangers.

Reply:
Done, see [300001e](https://github.com/ad-freiburg/qlever/pull/1058/commits/300001e739388e2dee1fee3120025a6ecaf00d6d).